src/client/fetch.ts

```diff
@@ -85,18 +89,89 @@ export class FetchRequest extends ChannelOwner<channels.FetchRequestChannel, cha
       const params = objectToArray(options.params);
       const method = options.method || request?.method();
       // Cannot call allHeaders() here as the request may be paused inside route handler.
-      const headersObj = options.headers || request?.headers() ;
+      let headersObj = options.headers || request?.headers() ;
+      let formData: channels.FormField[] | undefined;
+      let postDataBuffer: Buffer | undefined;
+      if (options.data) {
+        if (isString(options.data)) {
```

Comment:
Let's do some helper functions, so that it's easier to port to other clients.

Reply:
Done.